ts/sagas/wallet.ts

```diff
@@ -125,14 +123,15 @@ function* fetchTransactions(
       yield put(transactionsFetched(response.value.data));
     } else if (response.status === 401) {
       // unauthorized -- try refreshing the token
-      yield call(fetchPagoPaToken, pagoPaClient);
+      yield call(fetchPagoPaToken, pagoPaClient, walletToken);
```

Comment:
when the child saga only calls a few methods, it would be better to pass the method, instead of the whole client (makes it easier to test).

Reply:
Wouldn't passing the single method make the code harder to maintain? For instance, if you have N functions that are calling `fetchPagoPaToken` and they all pass `pagoPaClient.getSession` as an argument instead of `pagoPaClient`, if at some point `getSession` changes (either in name, or in required parameters), you need to update N places. If, on the other hand, you hand `fetchPagoPaToken` the client, you can access any and all methods you need in one place. 

Test-wise, shouldn't creating a mocked `PagoPaClient` + checking that `getSession` has been called once suffice?

Reply:
not sure what you mean... the problem of having to change the code that interacts with a method if the method changes, doesn't go await if you pass an object that provides that method - changing the method provided by that client implies that you will still have to change all the code that calls it

passing a method instead of a whole client makes testing much easier, instead of having to mock an object, you can pass any method - also decouples the code more

it's a matter of preferences, anyhow, at least don't pass the token, just pass the client instance

Reply:
I'll try to make my point here, though I understand that it's a matter of preferences. 

What is currently being done in the code can be summed up by:
```typescript
type PagoPaClient = {
  ...
  getSession: () => void
};

const client: PagoPaClient = ({
  getSession: () => "session"
});

const fetchTransaction = (pagoPaClient: PagoPaClient) => {
  ... do stuff with pagoPaClient.fetchTransaction() ...
  const result = yield call(fetchPagoPaToken, pagoPaClient);
  ...
}

const fetchWallet = (pagoPaClient: PagoPaClient) => {
  ... do stuff with pagoPaClient.fetchWallet() ...
  const result = yield call(fetchPagoPaToken, pagoPaClient);
  ...
}

const doSomethingElse = (pagoPaClient: PagoPaClient) => {
  ... do stuff with pagoPaClient.doSomethingElse() ...
  const result = yield call(fetchPagoPaToken, pagoPaClient);
  ...
}

const fetchPagoPaToken = (pagoPaClient: PagoPaClient) => {
  const session = yield call(pagoPaClient.getSession);
  ...
}
```

From what I gather, what you suggest is doing the following:
```typescript
type PagoPaClient = {
  ...
  getSession: () => void
};

const client: PagoPaClient = ({
  getSession: () => "session"
});

const fetchTransaction = (pagoPaClient: PagoPaClient) => {
  ... do stuff with pagoPaClient.fetchTransaction() ...
  const result = yield call(fetchPagoPaToken, pagoPaClient.getSession); // <==
  ...
}

const fetchWallet = (pagoPaClient: PagoPaClient) => {
  ... do stuff with pagoPaClient.fetchWallet() ...
  const result = yield call(fetchPagoPaToken, pagoPaClient.getSession); // <==
  ...
}

const doSomethingElse = (pagoPaClient: PagoPaClient) => {
  ... do stuff with pagoPaClient.doSomethingElse() ...
  const result = yield call(fetchPagoPaToken, pagoPaClient.getSession); // <==
  ...
}

const fetchPagoPaToken = (getSession: () => void) => { // <==
  const session = yield call(getSession); // <==
  ...
}
```
(the `<==` highlight the lines that have been changed)

Now, if `PagoPaClient`'s `getSession` were to change (e.g. renaming it to `getCurrentSession`), the first scenario would only require updating `fetchPagoPaToken`'s reference to `getSession`, while the second scenario would require changing the references in `fetchTransaction`, `fetchWallet` and `doSomethingElse`. 

I know refactoring tools make it easy to automate this kind of operations, but this is still introducing in the caller knowledge of what is being done by the callee